src/client/datascience/interactive-common/notebookProvider.ts

```diff
@@ -66,7 +66,7 @@ export class NotebookProvider implements INotebookProvider {
             if (
                 !(await this.jupyterNotebookProvider.connect({
                     resource: options.resource,
-                    token: options.token,
+                    tokenSource: options.tokenSource,
```

Comment:
I missed this. Why are we passing the source object. The lower layers should not be able to cancel it self (via the token), only the calling code should be able to cancel the lower calls.

Reply:
No that's not true. For window.withProgress to work, it needs to call cancel. It isn't passed a cancel token, it passes out a cancel token to the progress function. We implement this progress function and we need to look at its cancel value. The tokenSources are passed into this progress reporter.

Reply:
THis isn't actually relevant to this refactor but it was mixed in with my change from the install stuff. It was relevant there as the cancel button wasn't doing anything (it didn't actually cancel the install of a module)

Reply:
Ok, looking at the code, in KernelProgressReporter, we aren't displayhing the cancel button at all.
I.e. the first for `window.withProgress` doesn't contain the value `cancellable`
see here 
```typescript
            void window.withProgress(
                { location: ProgressLocation.Notification, title },
```

Reply:
Ie. this change isn't going to do anything.
Also, do we want a cancel button? I'm not against it, feels ok, but is a UX change and wanted to double check you wanted that.

Reply:
Should be removed now. I'll figure out something different when I implement the install change.